### DISCUSSION TITLE
Custom Datasets for audio/video files?

### DISCUSSION THREAD

**Discussion by andrewcstewart** [kedro-org/kedro#871] [Idea]
Hi folks.  Is anyone aware of any custom datasets out there for audio and/or video files?  I suspect I'll have to create my own classes for that, but thought I'd check to see if anyone is aware of existing work.

As for writing some new custom datasets, I'd be happy to hear any advice/suggestions from folks here.  I am thinking that the closest examples to model off of would be `kedro.extras.datasets.pillow.ImageDataSet`.

**Comment by mzjp2** [comment]
Closest thing I could find is: https://stackoverflow.com/questions/65900415/how-do-i-add-a-directory-of-wav-files-to-the-kedro-data-catalogue

**Reply by datajoely** [reply]
Hi @andrewcstewart this doesn't exist today - but we'd love to accept a PR for this. You're completely right that the [pillow.ImageDataSet](https://kedro.readthedocs.io/en/stable/_modules/kedro/extras/datasets/pillow/image_dataset.html#ImageDataSet) is the best analogue for what you're trying to do.

Instead of Pillow I think the best candidates look like:

- [MoviePy](https://zulko.github.io/moviepy/install.html)
- [ImageIO](https://imageio.readthedocs.io/en/stable/)
- [SimpleCV](http://simplecv.org/)
- [OpenCV2](https://pypi.org/project/opencv-python/)

Depending on what you need to do - this section of the [MoviePy docs](https://zulko.github.io/moviepy/getting_started/quick_presentation.html#advantages-and-limitations) may be helpful

**Comment by astrojuanlu** [comment]
These days there's actually a `VideoDataset` https://docs.kedro.org/projects/kedro-datasets/en/kedro-datasets-5.1.0/api/kedro_datasets.video.VideoDataset.html
